contrib/http_util/src/main/java/io/opencensus/contrib/http/util/HttpClientMetricGranularity.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.contrib.http.util;
+
+/** Enum class that determines the granularity at which opencensus http metrics are emitted. */
+public enum HttpClientMetricGranularity {
```

Comment:
Not sure why do you need this? What you can do is to always add the HOST tag, but don't change the default views. So you can define your own custom views that include the host :)

Reply:
what if some applications have a high cardinality hostname space, for example  web crawlers? Metric backends may run into cardinality issues in that case. What do you think?

Reply:
If the defined views do not include the HOST we just ignore the tag so no cardinality issue.

Reply:
I don't quite get your comment on custom views.